### PR TITLE
added controller version and logic to pull controller for functest

### LINKED ISSUE #98: Cannot find source file /var/folders/.../exe/appsody-controller to copy during test
**Describe the bug**
During `make test` i receive the following error:
```
[Debug] Attempting to load the controller from /Users/mhmxs/.appsody/appsody-controller
[Debug] Didn't find the controller in .appsody - copying from the binary directory...
[Debug] Binary location /var/folders/z2/f__84fm11736mw001xd06r4m0000gn/T/go-build846635544/b001/exe
[Debug] Attempting to copy the source controller from: /var/folders/z2/f__84fm11736mw001xd06r4m0000gn/T/go-build846635544/b001/exe/appsody-controller
[Error] Cannot find source file /var/folders/z2/f__84fm11736mw001xd06r4m0000gn/T/go-build846635544/b001/exe/appsody-controller to copy
```

**To Reproduce**
1. Remove `~/.appsody/appsody-controller`
2. Execute `make test`

**Expected behavior**
The test must compile on every machine not just where `~/.appsody/appsody-controller` exists.

**Actual behaviour**
Doesn't compile because `make test` doesn't create `appsody-controller` during the test.

**Environment Details (please complete the following information):**
 - OS: macOS 10.14.6 

**If applicable please specify:**
 - CLI version: latest master

### PR DESCRIPTION
**Pull request by kewegner**
Fixes #98 
Also fixes PR 181

Addes logic to choose appsody version of controller if the controller is not present during functest.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,12 +33,13 @@ package_binary = $(COMMAND)$(BINARY_EXT_$(os))
 all: lint test package ## Run lint, test, build, and package
 
 PHONY: install-controller
```
Comment (henrynash) [inline Makefile]: is "PHONY" a real label, as opposed to ".PHONY"? Or is this a typo in a previous merged commit by someone?

Comment (kewegner) [reply Makefile]: Its in the prior code drop.  Will look into it.

Comment (kewegner) [reply Makefile]: I changed it, for some reason it didn't seem to have any ill effects.  But to be consistent it should be .PHONY.

**Comment by kewegner** [comment]
tnixa has reviewed merging.

**Pull request merged**